priam/src/main/java/com/netflix/priam/backupv2/BackupVerificationTask.java

```diff
@@ -84,23 +84,24 @@ public void execute() throws Exception {
         Instant slo =
                 now.minus(backupRestoreConfig.getBackupVerificationSLOInHours(), ChronoUnit.HOURS);
         DateRange dateRange = new DateRange(slo, now);
-        List<BackupVerificationResult> verificationResults =
+        List<BackupMetadata> verifiedBackups =
                 backupVerification.verifyBackupsInRange(
                         BackupVersion.SNAPSHOT_META_SERVICE, dateRange);
 
-        verificationResults.forEach(
-                result -> {
-                    logger.info(
-                            "Sending {} message for backup: {}",
-                            AbstractBackupPath.BackupFileType.SNAPSHOT_VERIFIED,
-                            result.snapshotInstant);
-                    backupNotificationMgr.notify(result.remotePath, result.snapshotInstant);
-                });
+        verifiedBackups
+                .stream()
+                .filter(result -> result.getLastValidated().toInstant().isAfter(now))
+                .forEach(
+                        result -> {
+                            logger.info(
+                                    "Sending {} message for backup: {}",
+                                    AbstractBackupPath.BackupFileType.SNAPSHOT_VERIFIED,
+                                    result.getSnapshotLocation());
+                            backupNotificationMgr.notify(
+                                    result.getSnapshotLocation(), result.getStart().toInstant());
+                        });
 
-        if (!backupVerification
-                .findLatestVerifiedBackup(
-                        BackupVersion.SNAPSHOT_META_SERVICE, false /* force */, dateRange)
-                .isPresent()) {
+        if (verifiedBackups.isEmpty()) {
```

Comment:
nice, this is the bit that fixes the actual bug 👍

Reply:
315 lines of making the change easy. 5 lines of making the easy change.